### PR TITLE
Implemented Integrity MonthChecker

### PR DESCRIPTION
**Pull request by grimes2**
MonthChecker added. #2012

- [x] Change in CHANGELOG.md described
- [x] Tests created for changes
- [x] Manually tested changed features in running JabRef

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+public class MonthChecker implements Checker {
+
+    private static final Predicate<String> ONLY_AN_INTEGER_AND_MONTH_NORMALIZED = Pattern
+            .compile("[1-9]|10|11|12|#jan#|#feb#|#mar#|#apr#|#may#|#jun#|#jul#|#aug#|#sep#|#oct#|#nov#|#dec#")
+            .asPredicate();
+    private static final Predicate<String> MONTH_NORMALIZED = Pattern
+            .compile("#jan#|#feb#|#mar#|#apr#|#may#|#jun#|#jul#|#aug#|#sep#|#oct#|#nov#|#dec#")
+            .asPredicate();
+
+    private final BibDatabaseContext bibDatabaseContextEdition;
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java]: Seems like a copy und paste thing, please rename the variable.

File: src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+    public MonthChecker(BibDatabaseContext bibDatabaseContext) {
+        this.bibDatabaseContextEdition = Objects.requireNonNull(bibDatabaseContext);
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java]: same as above

File: src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import net.sf.jabref.logic.integrity.IntegrityCheck.Checker;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.database.BibDatabaseContext;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+public class MonthChecker implements Checker {
+
+    private static final Predicate<String> ONLY_AN_INTEGER_AND_MONTH_NORMALIZED = Pattern
+            .compile("[1-9]|10|11|12|#jan#|#feb#|#mar#|#apr#|#may#|#jun#|#jul#|#aug#|#sep#|#oct#|#nov#|#dec#")
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java]: Please use here only a pattern with an integer. As you have aready a second pattern for the normalized month you can simply combine the check.

File: src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        Optional<String> value = entry.getField(FieldName.MONTH);
+        if (!value.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        //BibLaTeX
+        if (bibDatabaseContextEdition.isBiblatexMode()
+                && !ONLY_AN_INTEGER_AND_MONTH_NORMALIZED.test(value.get().trim())) {
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/MonthChecker.java]: Here you can check with an || if the first pattern matches  or the second pattern matches

**Comment by grimes2** [comment]
pattern thing fixed

**Comment by Siedlerchr** [comment]
Thanks for your contribution! Looks good now :+1:  I will merge it in now

**Pull request merged**